### ISSUE TITLE
Use `Section` and `Heading` components inside `TearsheetShell` for modal header

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-products#8376]
We should avoid hard coding any heading levels in our components, these should always use the section and heading components from `@carbon/react`. The `TearsheetShell` specifies heading levels [here](https://github.com/carbon-design-system/ibm-products/blob/main/packages/ibm-products/src/components/Tearsheet/TearsheetShell.tsx#L500-L511). This should be refactored to avoid potential accessibility issues.

### Acceptance criteria
- [ ] Refactor the places where we hard code `h2` and `h3` within `TearsheetShell.tsx` with `Section` and `Heading` components
  - [h2 specified here](https://github.com/carbon-design-system/ibm-products/blob/main/packages/ibm-products/src/components/Tearsheet/TearsheetShell.tsx#L500-L502)
  - [h3 specified here](https://github.com/carbon-design-system/ibm-products/blob/main/packages/ibm-products/src/components/Tearsheet/TearsheetShell.tsx#L503-L511)
  - Composed [modal header](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/ComposedModal/ModalHeader.tsx#L125-L127) uses `h2` for both the label and header elements
